File: ruby/testcode/pathtraver_036.rb
require_relative 'shared'

FILEMAP = { 1 => '/app/files/report.pdf', 2 => '/app/files/invoice.pdf' }.freeze

# vuln-code-snippet start ruby_pt_id_lookup
def read_by_id(req)
  path = FILEMAP.fetch(req.param('id').to_i)
  BenchmarkResponse.ok(File.read(path)) # vuln-code-snippet safe-line ruby_pt_id_lookup
end
# vuln-code-snippet end ruby_pt_id_lookup

File: rust/testcode/pathtraver_036.rs
//! CWE-22: Helper function ignores its argument and returns a hardcoded safe path.

// vuln-code-snippet start testcodePathtraver036
fn get_file_path(_user_input: &str) -> &'static str {
    "/app/static/public.html"
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let input = req.param("f");

    match std::fs::read_to_string(get_file_path(&input)) { // vuln-code-snippet target-line testcodePathtraver036
        Ok(content) => super::shared::BenchmarkResponse::ok(&content),
        Err(e) => super::shared::BenchmarkResponse::error(&e.to_string()),
    }
}
// vuln-code-snippet end testcodePathtraver036
